ruby_on_rails/cucumber.md

```diff
@@ -0,0 +1,86 @@
+# Cucumber
+
+[Cucumber](https://cucumber.io) is a testing framework which allows us to specify feature tests using plain language.
+In our projects, we use Cucumber for e2e tests. We integrate it using the
+`cucumber-rails` gem.
+
+## Installing `cucumber-rails`
+
+The installation process of `cucumber-rails` is also documented on its
+[Github page](https://github.com/cucumber/cucumber-rails).
+
+To install `cucumber-rails`, add the following gems to your Gemfile:
+
+```rb
+group :test do
+    gem 'capybara', require: false
+    gem 'cucumber-rails', require: false
+    gem 'database-cleaner'
+end
+```
+
+Note that `capybara` and `database-cleaner` are optional. Most probably, you are going to need them.
+
+Run `bundle install` to install the gems.
+
+Finally, set up Cucumber with
+
+```sh
+rails generate cucumber:install
+```
+
+At this point, it might be a good idea to inspect the generated files and commit. At least, ckeck if you run into any
+rubocop issues. Probably, you have to add an exception in `.rubocop.yml`:
+
+```yaml
+AllCops:
+  Exclude:
+    - 'lib/tasks/cucumber.rake'
+```
+
+## Running your first Cucumber test
+
+Add the following files:
```

Comment:
I'd make two suggestions: one for new projects, where checking the home_controller.rb is a good idea. One for existing projects where I'd not add a controller just for that but I'd rather test an existing page.

Reply:
In this case, I'd just add a sentence like

> If you add Cucumber to an existing project, test a real page instead of using the given example test.

Reply:
perfect.